Git Streak Tracker/SettingsView.swift

```diff
@@ -10,29 +10,94 @@ import SwiftUI
 import WidgetKit
 import SwiftUIFontIcon
 
+
 struct SettingsView: View {
     @EnvironmentObject private var userStore: UserStore
+    @EnvironmentObject private var viewStore: ViewStore
     @State private var inputValue: String = ""
     @State private var isEditing = false
     
     let debouncer = Debouncer(delay: 0.5)
     
+    // Actions
     func loadInputField() {
         inputValue = userStore.username
     }
     
-    func storeUsername() {
-        userStore.username = inputValue // everytime username changes, the contributions are requested
+    func handleInputChanged() {
+        inputValue = inputValue.trimmingCharacters(in: .whitespacesAndNewlines)
```

Comment:
Can use `isValidGHUsername` here and then use the result to both
1. set or clear `userStore.error`
2. trigger debounced call to make request

Reply:
We might need one more flag in the userStore to mark a username as `valid` after we make a successful request.